src/app/clusters/icd-management-server/icd-management-cluster.cpp

```diff
@@ -0,0 +1,391 @@
+using namespace chip::Access;
+
+using chip::Protocols::InteractionModel::Status;
+
+namespace chip::app::Clusters {
+
+DataModel::ActionReturnStatus ICDManagementCluster::ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                                  AttributeValueEncoder & aEncoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case IcdManagement::Attributes::ClusterRevision::Id:
+        return aEncoder.Encode(kRevision);
+
+    case IcdManagement::Attributes::FeatureMap::Id:
+        return aEncoder.Encode(mFeatureMap);
+
+    case IcdManagement::Attributes::IdleModeDuration::Id:
+        return aEncoder.Encode(mICDConfigurationData.GetIdleModeDuration().count());
+
+    case IcdManagement::Attributes::ActiveModeDuration::Id:
+        return aEncoder.Encode(mICDConfigurationData.GetActiveModeDuration().count());
+
+    case IcdManagement::Attributes::ActiveModeThreshold::Id:
+        return aEncoder.Encode(mICDConfigurationData.GetActiveModeThreshold().count());
+
+#if CHIP_CONFIG_ENABLE_ICD_CIP
+    case IcdManagement::Attributes::RegisteredClients::Id:
+        return ReadRegisteredClients(aEncoder);
+
+    case IcdManagement::Attributes::ICDCounter::Id:
+        return aEncoder.Encode(mICDConfigurationData.GetICDCounter().GetValue());
+
+    case IcdManagement::Attributes::ClientsSupportedPerFabric::Id:
+        return aEncoder.Encode(mICDConfigurationData.GetClientsSupportedPerFabric());
+
+    case IcdManagement::Attributes::MaximumCheckInBackOff::Id:
+        return aEncoder.Encode(mICDConfigurationData.GetMaximumCheckInBackoff().count());
+#endif // CHIP_CONFIG_ENABLE_ICD_CIP
+
+#if CHIP_CONFIG_ENABLE_ICD_LIT
+    case IcdManagement::Attributes::OperatingMode::Id:
+        return ReadOperatingMode(aEncoder);
+#endif // CHIP_CONFIG_ENABLE_ICD_LIT
+
+    case IcdManagement::Attributes::UserActiveModeTriggerHint::Id:
+        return aEncoder.Encode(mUserActiveModeTriggerBitmap);
+
+    case IcdManagement::Attributes::UserActiveModeTriggerInstruction::Id:
+        return aEncoder.Encode(CharSpan(mUserActiveModeTriggerInstruction,
+                                        strnlen(mUserActiveModeTriggerInstruction, kUserActiveModeTriggerInstructionMaxLength)));
+
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+}
+
+CHIP_ERROR ICDManagementCluster::Attributes(const ConcreteClusterPath & path,
+                                            ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
+{
+    AttributeListBuilder attributeListBuilder(builder);
+    const DataModel::AttributeEntry kMandatoryAttributes[] = {
+        IcdManagement::Attributes::IdleModeDuration::kMetadataEntry,
+        IcdManagement::Attributes::ActiveModeDuration::kMetadataEntry,
+        IcdManagement::Attributes::ActiveModeThreshold::kMetadataEntry,
+    };
+    const bool hasCIP = mFeatureMap.Has(IcdManagement::Feature::kCheckInProtocolSupport);
```

Comment:
some bits are under `#if CHIP_CONFIG_ENABLE_ICD_CIP`

Should these be the same?